instrumentation/http/src/main/java/brave/http/HttpTracing.java

```diff
@@ -243,4 +252,32 @@ public HttpTracing build() {
       return new HttpTracing(this);
     }
   }
+
+  // volatile for get visibility. writes guarded by HttpTracing.class. Object to ensure unloadable
+  static volatile Object current = null;
+
+  /**
+   * Returns the most recently created tracing component iff it hasn't been closed. null otherwise.
+   *
+   * <p>This object should not be cached.
+   */
+  @Nullable public static HttpTracing current() {
+    return (HttpTracing) current;
+  }
+
+  private void maybeSetCurrent() {
+    if (current != null) return;
```

Comment:
Is it simpler to use `AtomicReference`? Pretty sure performance of `current` is the same (inlines the volatile read in the reference), and performance of the other methods (not that it matters) is better with less code.

```java
current.compareAndSet(null, this);
```

```java
current.compareAndSet(this, null);
```

Reply:
wonderful